packages/vsce/src/commands/showLogsCommand.ts

```diff
@@ -0,0 +1,122 @@
+      }
+    }
+  }
+
+  if (matchingProfiles.length > 0) {
+    return matchingProfiles[0];
+  }
+
+  // we couldn't find anything within a profile group
+  // filter down to just profiles that have the same hostname as our cics connection
+  const cicsHostProfiles = zosProfiles.filter((profile) => cicsProfile.profile.host === profile.profile.host);
+
+  return cicsHostProfiles[0];
+}
+
+function promptUserForProfile(zosProfiles: IProfileLoaded[]): Thenable<string> {
+  const profileNames = zosProfiles.map((value) => {
+    return value.name;
+  });
+
+  if (profileNames.length > 0) {
+    const quickPickOptions: vscode.QuickPickOptions = {
+      placeHolder: vscode.l10n.t("Select a profile to access the logs"),
+      ignoreFocusOut: true,
+      canPickMany: false,
+    };
+    return Gui.showQuickPick(profileNames, quickPickOptions);
+  } else {
+    return null;
+  }
+}
+
+export async function getJobIdForRegion(selectedRegion: CICSRegionTree): Promise<string> {
+  // when we have a CICSRGN table we have jobid, but not when we have a
+  // MAS table. we get either CICSRGN or MAS depending on where we are in the
+  // tree. request CICSRGN if jobid isn't available.
+  let jobid: string = selectedRegion.region.jobid;
+  if (!jobid) {
+    let response: ICMCIApiResponse;
+    try {
+      response = await getResource(selectedRegion.parentSession.getSession(), {
+        name: "CICSRegion",
+        cicsPlex: selectedRegion.parentPlex.plexName,
+        regionName: selectedRegion.region.cicsname,
+      });
+      jobid = response.response.records.cicsregion.jobid;
+    } catch (ex) {
+      console.log(ex);
+      // unlikely to get here but logging this would be useful in future
+    }
+  }
+  return jobid;
+}
+
+export function getShowRegionLogs(treeview: TreeView<any>) {
+  return commands.registerCommand("cics-extension-for-zowe.showRegionLogs", async (node) => {
+    const allSelectedRegions = findSelectedNodes(treeview, CICSRegionTree, node);
+    if (!allSelectedRegions || !(allSelectedRegions.length == 1)) {
+      await window.showErrorMessage("One CICS region must be selected");
+      return;
+    }
+    const selectedRegion: CICSRegionTree = allSelectedRegions[0];
+    const zosProfiles = (await ProfileManagement.getProfilesCache().fetchAllProfiles()).filter(
+      (profile) => profile.type === "zosmf" || profile.type === "rse",
+    );
+
+    let chosenProfileName: string;
+
+    // find profiles that match by base profile or hostname
+    const matchingZosProfiles = await findRelatedZosProfiles(selectedRegion.parentSession.profile, zosProfiles);
+    if (matchingZosProfiles) {
+      chosenProfileName = matchingZosProfiles.name;
+    } else {
+      // we couldn't find a matching profile - prompt the user with all zos profiles
+      chosenProfileName = await promptUserForProfile(zosProfiles);
+      if (chosenProfileName === null) {
+        window.showErrorMessage("Could not find any z/OSMF or RSE profiles.");
```

Comment:
It's possible this code can be reached if the user is prompted to choose a profile and presses `Esc` so the error message isn't too relevant (I think). We could see what behaviour we do in other places where quickpicks can be cancelled.

Reply:
I tested this - thanks for the thought. If you hit escape, the answer is undefined, which isn't null.

So the error message doesn't show, which is good. The behaviour I think is bang on what we'd want. However the undefined vs null seems a bit "luck rather than judgement" ... I'm thinking perhaps I add a comment to the method explaining the possible returns 🤔